pkg/kubelet/pleg/generic.go

```diff
@@ -255,10 +255,34 @@ func (g *GenericPLEG) Relist() {
 	for pid := range g.podRecords {
 		oldPod := g.podRecords.getOld(pid)
 		pod := g.podRecords.getCurrent(pid)
+
+		var cachePodStatus *kubecontainer.PodStatus
+		var podStatus *kubecontainer.PodStatus
+
+		// get cachePodStatus and runtimePodStatus to help distinguish the resized pod.
+		if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
+			if !g.cacheEnabled() {
+				klog.ErrorS(err, "GenericPLEG: Cache is not enabled")
+				return
+			}
+			cachePodStatus, err = g.cache.Get(pid)
+			if err != nil {
+				klog.ErrorS(err, "GenericPLEG: Unable to retrieve pods", "pid", pid)
+				return
+			}
+			if pod != nil {
+				podStatus, err = g.runtime.GetPodStatus(ctx, pod.ID, pod.Name, pod.Namespace)
```

Comment:
If I'm reading this correctly, this is going to call `GetPodStatus()` on every pod, on ever relist loop? Won't that make the relist too expensive? I assume the logic that made this conditional originally was intentional to avoid this.

I think this needs a lot of scale & performance testing before we can proceed with this change.